src/core/communication.cpp

```diff
@@ -483,6 +483,22 @@ void mpi_set_time_step_slave(double dt) {
 REGISTER_CALLBACK(mpi_set_time_step_slave)
 
 void mpi_set_time_step(double time_s) {
+  if (time_s <= 0.)
+    throw std::invalid_argument("time_step must be > 0.");
+  if (lb_lbfluid_get_lattice_switch() !=  ActiveLB::NONE){
```

Comment:
could you please move this to a free function and correctly ifdef for the LB feature.

Reply:
there is no `LB` macro, could you precise what you meant?

Reply:
Ah right, I forgot